doltpy/cli/dolt.py

```diff
@@ -133,30 +127,37 @@ def __init__(self, name: str, url: str):
         self.url = url
 
 
-class ServerConfig:
+class DoltHubContext:
+
     def __init__(self,
-                 config: str = None,
-                 host: str = DEFAULT_HOST,
-                 port: int = DEFAULT_PORT,
-                 user: str = None,
-                 password: str = None,
-                 timeout: int = None,
-                 readonly: bool = None,
-                 loglevel: str = None,
-                 multi_db_dir: str = None,
-                 no_auto_commit: bool = None,
-                 echo: bool = False):
-        self.config = config
-        self.host = host
-        self.port = port
-        self.user = user
-        self.password = password
-        self.timeout = timeout
-        self.readonly = readonly
-        self.loglevel = loglevel
-        self.multi_db_dir = multi_db_dir
-        self.no_auto_commit = no_auto_commit
-        self.echo = echo
+                 db_path: str,
+                 path: str = None,
+                 remote: str = 'origin',
+                 tables_to_read: List[str] = None):
+        self.db_path = db_path
+        self.path = tempfile.mkdtemp() if not path else path
+        self.remote = remote
+        self.dolt = None
+        self.tables_to_read = tables_to_read
+
+    def __enter__(self):
+        try:
+            dolt = Dolt(self.path)
+            logger.info(f'Dolt database found at path provided ({self.path}), pulling from remote "{self.remote}"')
+            dolt.pull(self.remote)
+        except AssertionError:
+            assert self.db_path is not None, 'Cannot clone remote data without db_path set'
+            if self.tables_to_read:
+                logger.info(f'Running read-tables, creating a fresh copy of {self.db_path}')
+                dolt = Dolt.read_tables(self.db_path, 'master', table_or_tables=self.tables_to_read)
+            else:
+                logger.info(f'Running clone, cloning remote {self.db_path}')
+                dolt = Dolt.clone(self.db_path, self.path)
+
+        self.dolt = dolt
+
+    def __exit__(self):
+        pass
 
 
 class Dolt:
```

Comment:
I can add type classes if its helpful, I hate mypy but it's hard to argue with the added code quality